lib/graphql/types/iso_8601_date_time.rb

```diff
@@ -52,8 +52,12 @@ def self.coerce_result(value, _ctx)
       def self.coerce_input(str_value, _ctx)
         Time.iso8601(str_value)
       rescue ArgumentError, TypeError
-        # Invalid input
-        nil
+        begin
+          DateTime.iso8601(str_value).to_time
```

Comment:
I think this should be `Date.iso8601(str_value).to_time` so it inherits the system timezone, similarly to how `coerce_result` handles `Date` objects. `DateTime` always defaults to UTC.

```
irb(main):001:0> require 'time'
=> true
irb(main):002:0> ENV['TZ'] = 'America/Toronto'
irb(main):003:0> str_value = '2018-06-07'
irb(main):004:0> DateTime.iso8601(str_value)
=> #<DateTime: 2018-06-07T00:00:00+00:00 ((2458277j,0s,0n),+0s,2299161j)>
irb(main):005:0> Date.iso8601(str_value)
=> #<Date: 2018-06-07 ((2458277j,0s,0n),+0s,2299161j)>
irb(main):006:0> DateTime.iso8601(str_value).to_time
=> 2018-06-07 00:00:00 +0000
irb(main):007:0> Date.iso8601(str_value).to_time
=> 2018-06-07 00:00:00 -0400
irb(main):008:0> ENV['TZ'] = 'Etc/UTC'
irb(main):009:0> DateTime.iso8601(str_value).to_time
=> 2018-06-07 00:00:00 +0000
irb(main):010:0> Date.iso8601(str_value).to_time
=> 2018-06-07 00:00:00 +0000
```

Reply:
Thanks for the suggestion :+1: !